include/net/http_parser.h

```diff
@@ -24,10 +24,12 @@
 extern "C" {
 #endif
 
+#if !defined(CONFIG_HTTP_PARSER_URL_ONLY)
```

Comment:
i remember having seen this change a while back wanted to comment on it but somehow forgot, dont really like all of this ifdefing, it would cleaner if we split the url parser out into its own library instead and reused it in the http library and elsewhere. All of those #ifs make the code and header unreadable